application/src/main/java/org/togetherjava/tjbot/features/help/HelpThreadCreatedListener.java

```diff
@@ -84,8 +84,11 @@ private boolean wasThreadAlreadyHandled(long threadChannelId) {
     private void handleHelpThreadCreated(ThreadChannel threadChannel) {
         threadChannel.retrieveStartMessage().flatMap(message -> {
             registerThreadDataInDB(message, threadChannel);
-            return generateAutomatedResponse(threadChannel);
-        }).flatMap(message -> pinOriginalQuestion(threadChannel)).queue();
+            return sendHelperHeadsUp(threadChannel)
+                .flatMap(any -> HelpThreadCreatedListener.isContextSufficient(message),
+                        any -> createAIResponse(threadChannel, message))
+                .flatMap(any -> pinOriginalQuestion(message));
```

Comment:
```suggestion
                .flatMap(Message::pin);
```

Reply:
This would change the behavior, because now it would pin the `any` Message from the `RestAction` call before, which in this case is the message returned by `createAIRsponse()`.

Reply:
But technically, I could do this:
```java
.flatMap(any -> message.pin());
```
This way I can remove the method as well

Reply:
Sry i mean `message::pin`

Reply:
well that doesnt work, it gives error, because it tries to give `any` as the method parameter:
![image](https://github.com/Together-Java/TJ-Bot/assets/88111627/37fb95ba-7abd-4bd5-a320-bd4b92b725c3)